File: client/src/types/setting.ts
export interface Setting {
    setting_id: number;
    setting_key: string;
    setting_value: string;
    description?: string;
    created_at: string;
    updated_at: string;
}

export interface CreateSettingRequest {
    setting_key: string;
    setting_value: string;
    description?: string;
}

export interface UpdateSettingRequest {
    setting_value: string;
}

File: wails-app/inventory-management/pkg/domain/setting.go
package domain

import "time"

// Setting represents a global application setting
type Setting struct {
	SettingID    int64     `json:"setting_id"`
	SettingKey   string    `json:"setting_key"`
	SettingValue string    `json:"setting_value"`
	Description  *string   `json:"description,omitempty"`
	CreatedAt    time.Time `json:"created_at"`
	UpdatedAt    time.Time `json:"updated_at"`
}
